### PR TITLE
[oneDPL][tests] + Add tests for some functions from std::cmath and std::limits

### PR DESCRIPTION
**Pull request by SergeyKopienko**
Which functions were tested:

- std::max;
- std::min;
- std::abs;
- std::ceil;
- std::copysign;
- std::copysignf;
- std::fabs;
- std::floor;
- std::fmax;
- std::fmaxf;
- std::fmin;
- std::fminf;
- std::isgreater;
- std::isgreaterequal;
- std::isinf;
- std::isless;
- std::islessequal;
- std::isnan;
- std::isunordered;
- std::nan;
- std::nanf;
- std::round;
- std::roundf;
- std::trunc;
- std::truncf; 

Where these functions were tested:
- on host;
- in Kernel.
  
Source tests was copied from:
- https://github.com/llvm/llvm-project/blob/main/libcxx/test/std/algorithms/alg.sorting/alg.min.max/max.pass.cpp
- https://github.com/llvm/llvm-project/blob/main/libcxx/test/std/algorithms/alg.sorting/alg.min.max/max_comp.pass.cpp
- https://github.com/llvm/llvm-project/blob/main/libcxx/test/std/algorithms/alg.sorting/alg.min.max/max_init_list_comp.pass.cpp
- https://github.com/llvm/llvm-project/blob/main/libcxx/test/std/algorithms/alg.sorting/alg.min.max/max_init_list.pass.cpp
- https://github.com/llvm/llvm-project/blob/main/libcxx/test/std/algorithms/alg.sorting/alg.min.max/min_comp.pass.cpp
- https://github.com/llvm/llvm-project/blob/main/libcxx/test/std/algorithms/alg.sorting/alg.min.max/min_init_list_comp.pass.cpp
- https://github.com/llvm/llvm-project/blob/main/libcxx/test/std/algorithms/alg.sorting/alg.min.max/min_init_list.pass.cpp
- https://github.com/llvm/llvm-project/blob/main/libcxx/test/std/algorithms/alg.sorting/alg.min.max/min.pass.cpp
- https://github.com/llvm/llvm-project/blob/main/libcxx/test/std/language.support/support.limits/limits/numeric.limits.members/infinity.pass.cpp
- https://github.com/llvm/llvm-project/blob/main/libcxx/test/std/language.support/support.limits/limits/numeric.limits.members/lowest.pass.cpp
- https://github.com/llvm/llvm-project/blob/main/libcxx/test/std/language.support/support.limits/limits/numeric.limits.members/max.pass.cpp
- https://github.com/llvm/llvm-project/blob/main/libcxx/test/std/language.support/support.limits/limits/numeric.limits.members/quiet_NaN.pass.cpp
- https://github.com/llvm/llvm-project/blob/main/libcxx/test/std/numerics/c.math/cmath.pass.cpp

### CODE REVIEW

**Comment by SergeyKopienko** [comment]
@MikeDvorskiy could you please take a look to this PR ?

File: test/xpu_api/language.support/support.limits/limits/infinity.pass.cpp [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+#endif
+    test<short>(0);
+    test<unsigned short>(0);
+    test<int>(0);
+    test<unsigned int>(0);
+    test<long>(0);
+    test<unsigned long>(0);
+    test<long long>(0);
+    test<unsigned long long>(0);
+#ifndef TEST_HAS_NO_INT128
```
Comment (MikeDvorskiy) [inline test/xpu_api/language.support/support.limits/limits/infinity.pass.cpp:54]: I suggest modifying it in a positive logic instead of double negation.

Comment (SergeyKopienko) [reply test/xpu_api/language.support/support.limits/limits/infinity.pass.cpp:54]: This logic was copied from LLVM tests...
My suggestion - not modify it.

File: test/parallel_api/algorithm/alg.sorting/alg.min.max/max.pass.cpp [hunk 2]
```diff
@@ -0,0 +1,58 @@
+//===----------------------------------------------------------------------===//
```
Comment (MikeDvorskiy) [inline test/parallel_api/algorithm/alg.sorting/alg.min.max/max.pass.cpp:1]: It looks that we have wrong file "header" here.

File: test/parallel_api/algorithm/alg.sorting/alg.min.max/max_comp.pass.cpp [hunk 3]
```diff
@@ -0,0 +1,60 @@
+//===----------------------------------------------------------------------===//
```
Comment (MikeDvorskiy) [inline test/parallel_api/algorithm/alg.sorting/alg.min.max/max_comp.pass.cpp:1]: It looks that we have wrong file "header" here.

File: test/parallel_api/algorithm/alg.sorting/alg.min.max/max_init_list.pass.cpp [hunk 4]
```diff
@@ -0,0 +1,46 @@
+//===----------------------------------------------------------------------===//
```
Comment (MikeDvorskiy) [inline test/parallel_api/algorithm/alg.sorting/alg.min.max/max_init_list.pass.cpp:1]: It looks that we have wrong file "header" here and everywhere below in new files

File: test/parallel_api/algorithm/alg.sorting/alg.min.max/max_comp.pass.cpp (hunk 3 above)
Comment (SergeyKopienko) [reply test/parallel_api/algorithm/alg.sorting/alg.min.max/max_comp.pass.cpp:2]: Fixed

File: test/parallel_api/algorithm/alg.sorting/alg.min.max/max_init_list.pass.cpp (hunk 4 above)
Comment (SergeyKopienko) [reply test/parallel_api/algorithm/alg.sorting/alg.min.max/max_init_list.pass.cpp:2]: Fixed.

**Review by MikeDvorskiy — APPROVED** [review]
LGTM

**Pull request merged**